### PR TITLE
Updating patient search

### PR DESCRIPTION
**Pull request by MadhurRaj**
Adding the new parameters supported for searching patients.

### CODE REVIEW

File: content/dstu2/patient.md [hunk 1]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
+- Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+- The `gender` parameter can only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
+- The `name`, `family`, and `given` parameters will search for current names only.
+- The `identifier`, `name`, or `family`, `given`, `phone`, `email`, `address-postalcode`, `gender` parameters may be provided exactly once and may have only a single value.
+- The `birthdate` parameter may be provided twice to indicate a date range, but must contain the inclusive prefixes ('le' and 'ge')
+- The `birthdate` parameter may be provided once with the following prefixes: 'ge', 'le', 'gt', 'lt', 'eq'
+- The `name`, `family`, and `given` parameters support the ':exact' modifier
+- The value of the `gender` search parameter must be part of the [administrative-gender] valueset.
+- If a system is passed in with the `phone` parameter it must be phone.
+- If a system is passed in with the `email` parameter it must be phone.
```
Comment (MaxPhilips) [inline content/dstu2/patient.md]: Must be email?

Comment (MadhurRaj) [inline content/dstu2/patient.md]: whoops, fixed

File: content/dstu2/patient.md [hunk 2]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
+ `family`                | This and/or any other search param, or `_id`        | [`string`] | The start of the family name of the patient. Example: `Adam`
+ `given`                 | This and/or any other search param, or `_id`        | [`string`] | The start of the given name of the patient. Example: `Tim`
+ `birthdate`             | This and/or any other search param, or `_id`        | [`date`]   | The patient's date of birth.  Example: `1961-01-16`
+ `phone`                 | This and/or any other search param, or `_id`        | [`token`]  | The patient's phone number. Example: `1111111111`
+ `email`                 | This and/or any other search param, or `_id`        | [`token`]  | The patient's email address. Example: `example@example.com`
+ `address-postalcode`    | This and/or any other search param, or `_id`        | [`string`] | The postal code in the address details of the patient. Example: `11111`
+ `gender`                | No, must be provided with any required search parameter except `_id` | [`string`] | The gender of the patient. Example: `male`
```
Comment (cerbs) [inline content/dstu2/patient.md]: This can be just "No". You already explain further in the Notes section

File: content/dstu2/patient.md [hunk 3]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
-
+- Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+- The `gender` parameter can only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
+- The `name`, `family`, and `given` parameters will search for current names only.
+- The `identifier`, `name`, or `family`, `given`, `phone`, `email`, `address-postalcode`, `gender` parameters may be provided exactly once and may have only a single value.
+- The `birthdate` parameter may be provided twice to indicate a date range, but must contain the inclusive prefixes ('le' and 'ge')
+- The `birthdate` parameter may be provided once with the following prefixes: 'ge', 'le', 'gt', 'lt', 'eq'
+- The `name`, `family`, and `given` parameters support the ':exact' modifier
+- The value of the `gender` search parameter must be part of the [administrative-gender] valueset.
```
Comment (cerbs) [inline content/dstu2/patient.md]: Shouldn't need to specify this since this is a required binding

File: content/dstu2/patient.md [hunk 4]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
+- If a system is passed in with the `phone` parameter it must be phone.
```
Comment (cerbs) [inline content/dstu2/patient.md]: Don't need these lines for phone or email.  These is expected behavior

File: content/dstu2/patient.md [hunk 5]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
-
-### Limitations
-
-- The `birthdate`, `name` and `telecom` search parameters are only available in the original sandbox (`https://fhir-*.sandboxcerner.com/dstu2/d075cf8b-3261-481d-97e5-ba6c48d3b41f/`) and will perform a "fuzzy" search.
-
+- Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+- The `gender` parameter can only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
+- The `name`, `family`, and `given` parameters will search for current names only.
+- The `identifier`, `name`, or `family`, `given`, `phone`, `email`, `address-postalcode`, `gender` parameters may be provided exactly once and may have only a single value.
+- The `birthdate` parameter may be provided twice to indicate a date range, but must contain the inclusive prefixes ('le' and 'ge')
```
Comment (cerbs) [inline content/dstu2/patient.md]: Remove the parentheses

File: content/dstu2/patient.md [hunk 6]
```diff
@@ -50,22 +50,30 @@ _Implementation Notes_
...
 
-- Either the `_id`, `identifier` or a combination of `birthdate`, `name` and `telecom` parameters must be provided.
-- The `identifier`, `birthdate`, `name`, or `telecom` parameters may be provided exactly once and may have only a single value.
-
-### Limitations
-
-- The `birthdate`, `name` and `telecom` search parameters are only available in the original sandbox (`https://fhir-*.sandboxcerner.com/dstu2/d075cf8b-3261-481d-97e5-ba6c48d3b41f/`) and will perform a "fuzzy" search.
-
+- Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+- The `gender` parameter can only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
```
Comment (cerbs) [inline content/dstu2/patient.md]: can => may

File: content/dstu2/patient.md [hunk 7]
```diff
@@ -109,3 +117,4 @@ The ability to perform patient combine or uncombine operations is not available
 [`_count`]: http://hl7.org/fhir/DSTU2/search.html#count
 [`number`]: http://hl7.org/fhir/DSTU2/search.html#number
 [Patient.animal]: http://hl7.org/fhir/DSTU2/patient-definitions.html#Patient.animal
+[administrative-gender]: http://hl7.org/fhir/valueset-administrative-gender.html
```
Comment (cerbs) [inline content/dstu2/patient.md]: Can remove given above comment

File: content/dstu2/patient.md [hunk 8]
```diff
@@ -50,22 +50,27 @@ _Implementation Notes_
...
-
-- The `birthdate`, `name` and `telecom` search parameters are only available in the original sandbox (`https://fhir-*.sandboxcerner.com/dstu2/d075cf8b-3261-481d-97e5-ba6c48d3b41f/`) and will perform a "fuzzy" search.
-
+- Either the `_id`, or a combination of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters must be provided.
+- The `gender` parameter may only be provided if at least one of `identifier` , `birthdate`, `name`, `given`, `family`, `address-postalcode`, `phone`, or `email` parameters is provided.
+- The `name`, `family`, and `given` parameters will search for current names only.
+- The `identifier`, `name`, or `family`, `given`, `phone`, `email`, `address-postalcode`, `gender` parameters may be provided exactly once and may have only a single value.
+- The `birthdate` parameter may be provided twice to indicate a date range, but must contain the inclusive prefixes 'le' and 'ge'
+- The `birthdate` parameter may be provided once with the following prefixes: 'ge', 'le', 'gt', 'lt', 'eq'
+- The `name`, `family`, and `given` parameters support the ':exact' modifier
```
Comment (cerbs) [inline content/dstu2/patient.md]: Combine this with line 69:
```
The `name`, `family`, and `given` parameters support the ':exact' modifier and will search for current names only.
```

**Comment by cerbs** [comment]
+1 otherwise

**Comment by MaxPhilips** [comment]
Can you move the or to the end of
"`identifier`, `name`, or `family`, `given`, `phone`, `email`, `address-postalcode`, `gender`"?

+1

**Comment by MadhurRaj** [comment]
Addressed all comments. Squashing and Merging.

**Pull request merged**